[stdlib] API additions for basic noncopyable primitives

lorentey opened the pull request:
This PR implements the stdlib API additions introduced by [SE-0437](https://github.com/apple/swift-evolution/blob/main/proposals/0437-noncopyable-stdlib-primitives.md):

- Add `exchange(_:with:)`
- Add `Optional.take()`
- Add `Unsafe[Mutable]BufferPointer.extracting(_:)`
- Finish generalizing `withExtendedLifetime`
- Radically simplify the implementation of `swap(_:_:)`

lorentey committed:
[stdlib] API additions for basic noncopyable primitives

- Add `exchange(_:with:)`
- Add `Optional.take()`
- Add `Unsafe[Mutable]BufferPointer.extracting(_:)`
- Finish generalizing `withExtendedLifetime`
- Radically simplify the implementation of `swap(_:_:)`

lorentey commented:
@swift-ci test

lorentey commented:
@swift-ci test source compatibility

lorentey commented:
@swift-ci build toolchain

Azoy commented on stdlib/public/core/Optional.swift:
```diff
@@ -418,12 +418,18 @@ extension Optional where Wrapped: ~Copyable {
     return result
   }
+
+  @available(*, deprecated, renamed: "take")
```
This was never public, I don't see why we need to deprecate it.

lorentey replied:
This is a conservative move, to avoid breaking any code (outside of this repo) that may have already adopted the `_take` spelling since it landed earlier this year.

lorentey replied:
Removed! There is little point to keeping it on this branch, and the risk is marginal even on release/6.0

lorentey committed:
[test] Update source stability list of false positives

lorentey commented:
@swift-ci test

lorentey commented:
CoreStore 5.0 failed on iOS and watchOS.

iOS:

```
Please submit a bug report (https://swift.org/contributing/#reporting-bugs) and include the crash backtrace.
Stack dump:
0.	Program arguments: /Users/ec2-user/jenkins/workspace-private/swift-PR-source-compat-suite-debug-macos/build/compat_macos/install/toolchain/usr/bin/swift-frontend -frontend -c -filelist "/Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/Build/Intermediates.noindex/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/Objects-normal/arm64/sources-1" -primary-file /Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/Sources/CoreStoreSchema.swift -primary-file /Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/Sources/CoreStoreBridge.swift -primary-file /Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/Sources/CSUnsafeDataTransaction.swift -primary-file /Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/Sources/NSManagedObjectContext+Querying.swift -primary-file /Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/Sources/InferredSchemaMappingProvider.swift -primary-file /Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/Sources/CoreStoreManagedObject.swift -primary-file /Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/Sources/FetchChainBuilder.swift -primary-file /Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/Sources/Relationship.swift -primary-file /Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/Sources/CoreStoreLogger.swift -primary-file /Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/Sources/NSManagedObjectModel+Migration.swift -primary-file /Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/Sources/CustomSchemaMappingProvider.swift -primary-file /Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/Sources/NSManagedObject+DynamicModel.swift -primary-file /Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/Sources/DispatchQueue+CoreStore.swift -supplementary-output-file-map "/Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/Build/Intermediates.noindex/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/Objects-normal/arm64/supplementaryOutputs-9" -target arm64-apple-ios16.0 -Xllvm -aarch64-use-tbi -enable-objc-interop -sdk /Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.2.sdk -I /Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/Build/Products/Debug-iphoneos -F /Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/Build/Products/Debug-iphoneos -no-color-diagnostics -application-extension -enable-testing -g -import-underlying-module -module-cache-path /Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/ModuleCache.noindex -swift-version 5 -enforce-exclusivity=checked -Onone -D DEBUG -serialize-debugging-options -const-gather-protocols-file "/Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/Build/Intermediates.noindex/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/Objects-normal/arm64/CoreStore iOS_const_extract_protocols.json" -enable-bare-slash-regex -empty-abi-descriptor -validate-clang-modules-once -clang-build-session-file /Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/ModuleCache.noindex/Session.modulevalidation -Xcc -working-directory -Xcc /Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore -resource-dir /Users/ec2-user/jenkins/workspace-private/swift-PR-source-compat-suite-debug-macos/build/compat_macos/install/toolchain/usr/lib/swift -enable-anonymous-context-mangled-names -Xcc "-I/Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/Build/Intermediates.noindex/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/swift-overrides.hmap" -Xcc -iquote -Xcc "/Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/Build/Intermediates.noindex/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/CoreStore-generated-files.hmap" -Xcc "-I/Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/Build/Intermediates.noindex/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/CoreStore-own-target-headers.hmap" -Xcc "-I/Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/Build/Intermediates.noindex/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/CoreStore-all-non-framework-target-headers.hmap" -Xcc -ivfsoverlay -Xcc "/Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/Build/Intermediates.noindex/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/all-product-headers.yaml" -Xcc -iquote -Xcc "/Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/Build/Intermediates.noindex/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/CoreStore-project-headers.hmap" -Xcc -I/Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/Build/Products/Debug-iphoneos/include -Xcc "-I/Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/Build/Intermediates.noindex/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/DerivedSources-normal/arm64" -Xcc "-I/Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/Build/Intermediates.noindex/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/DerivedSources/arm64" -Xcc "-I/Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/Build/Intermediates.noindex/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/DerivedSources" -Xcc -DDEBUG=1 -Xcc -ivfsoverlay -Xcc "/Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/Build/Intermediates.noindex/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/unextended-module-overlay.yaml" -module-name CoreStore -frontend-parseable-output -disable-clang-spi -target-sdk-version 17.2 -target-sdk-name iphoneos17.2 -external-plugin-path /Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.2.sdk/usr/lib/swift/host/plugins#/Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.2.sdk/usr/bin/swift-plugin-server -external-plugin-path /Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.2.sdk/usr/local/lib/swift/host/plugins#/Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.2.sdk/usr/bin/swift-plugin-server -external-plugin-path /Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/lib/swift/host/plugins#/Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin/swift-plugin-server -external-plugin-path /Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/local/lib/swift/host/plugins#/Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin/swift-plugin-server -plugin-path /Users/ec2-user/jenkins/workspace-private/swift-PR-source-compat-suite-debug-macos/build/compat_macos/install/toolchain/usr/lib/swift/host/plugins -plugin-path /Users/ec2-user/jenkins/workspace-private/swift-PR-source-compat-suite-debug-macos/build/compat_macos/install/toolchain/usr/local/lib/swift/host/plugins -o "/Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/Build/Intermediates.noindex/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/Objects-normal/arm64/CoreStoreSchema.o" -o "/Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/Build/Intermediates.noindex/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/Objects-normal/arm64/CoreStoreBridge-84979716fa621444a9d7ed53d0272860.o" -o "/Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/Build/Intermediates.noindex/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/Objects-normal/arm64/CSUnsafeDataTransaction.o" -o "/Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/Build/Intermediates.noindex/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/Objects-normal/arm64/NSManagedObjectContext+Querying.o" -o "/Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/Build/Intermediates.noindex/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/Objects-normal/arm64/InferredSchemaMappingProvider.o" -o "/Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/Build/Intermediates.noindex/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/Objects-normal/arm64/CoreStoreManagedObject.o" -o "/Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/Build/Intermediates.noindex/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/Objects-normal/arm64/FetchChainBuilder.o" -o "/Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/Build/Intermediates.noindex/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/Objects-normal/arm64/Relationship.o" -o "/Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/Build/Intermediates.noindex/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/Objects-normal/arm64/CoreStoreLogger.o" -o "/Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/Build/Intermediates.noindex/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/Objects-normal/arm64/NSManagedObjectModel+Migration.o" -o "/Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/Build/Intermediates.noindex/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/Objects-normal/arm64/CustomSchemaMappingProvider.o" -o "/Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/Build/Intermediates.noindex/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/Objects-normal/arm64/NSManagedObject+DynamicModel.o" -o "/Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/Build/Intermediates.noindex/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/Objects-normal/arm64/DispatchQueue+CoreStore.o" -index-unit-output-path "/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/Objects-normal/arm64/CoreStoreSchema.o" -index-unit-output-path "/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/Objects-normal/arm64/CoreStoreBridge-84979716fa621444a9d7ed53d0272860.o" -index-unit-output-path "/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/Objects-normal/arm64/CSUnsafeDataTransaction.o" -index-unit-output-path "/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/Objects-normal/arm64/NSManagedObjectContext+Querying.o" -index-unit-output-path "/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/Objects-normal/arm64/InferredSchemaMappingProvider.o" -index-unit-output-path "/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/Objects-normal/arm64/CoreStoreManagedObject.o" -index-unit-output-path "/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/Objects-normal/arm64/FetchChainBuilder.o" -index-unit-output-path "/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/Objects-normal/arm64/Relationship.o" -index-unit-output-path "/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/Objects-normal/arm64/CoreStoreLogger.o" -index-unit-output-path "/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/Objects-normal/arm64/NSManagedObjectModel+Migration.o" -index-unit-output-path "/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/Objects-normal/arm64/CustomSchemaMappingProvider.o" -index-unit-output-path "/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/Objects-normal/arm64/NSManagedObject+DynamicModel.o" -index-unit-output-path "/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/Objects-normal/arm64/DispatchQueue+CoreStore.o"
1.	Apple Swift version 6.0-dev (LLVM d5c188027e398b3, Swift e3af34a6eacf2d5)
2.	Compiling with effective version 5.10
3.	Contents of /Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/build/Build/Intermediates.noindex/CoreStore.build/Debug-iphoneos/CoreStore iOS.build/Objects-normal/arm64/sources-1:
[...]
4.	While evaluating request IRGenRequest(IR Generation for file "/Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/Sources/Relationship.swift")
5.	While emitting IR SIL function "@$s9CoreStore21RelationshipContainerO13ToManyOrderedCyqd__Sicig".
 for getter for subscript(_:) (at /Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/Sources/Relationship.swift:903:12)
Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
0  swift-frontend           0x0000000114516dd8 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) + 40
1  swift-frontend           0x0000000114514f78 llvm::sys::RunSignalHandlers() + 248
2  swift-frontend           0x000000011451743e SignalHandler(int) + 270
3  libsystem_platform.dylib 0x00007ff817fa05ed _sigtramp + 29
4  swift-frontend           0x00000001144ce21e llvm::Twine::str() const + 430
5  swift-frontend           0x000000010e4c9563 (anonymous namespace)::SyncCallEmission::emitCallToUnmappedExplosion(llvm::CallBase*, swift::irgen::Explosion&) + 531
6  swift-frontend           0x000000010e4b237d swift::irgen::CallEmission::emitToExplosion(swift::irgen::Explosion&, bool) + 1037
7  swift-frontend           0x000000010e6c332b (anonymous namespace)::IRGenSILFunction::visitFullApplySite(swift::FullApplySite) + 4363
8  swift-frontend           0x000000010e69df1c (anonymous namespace)::IRGenSILFunction::visitSILBasicBlock(swift::SILBasicBlock*) + 1692
9  swift-frontend           0x000000010e69c73d (anonymous namespace)::IRGenSILFunction::emitSILFunction() + 11389
10 swift-frontend           0x000000010e69948d swift::irgen::IRGenModule::emitSILFunction(swift::SILFunction*) + 2253
11 swift-frontend           0x000000010e518084 swift::irgen::IRGenerator::emitGlobalTopLevel(std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>> const&) + 836
12 swift-frontend           0x000000010e63dc40 swift::IRGenRequest::evaluate(swift::Evaluator&, swift::IRGenDescriptor) const + 2720
13 swift-frontend           0x000000010e698701 swift::GeneratedModule swift::SimpleRequest<swift::IRGenRequest, swift::GeneratedModule (swift::IRGenDescriptor), (swift::RequestFlags)9>::callDerived<0ul>(swift::Evaluator&, std::__1::integer_sequence<unsigned long, 0ul>) const + 209
14 swift-frontend           0x000000010e6985ee swift::SimpleRequest<swift::IRGenRequest, swift::GeneratedModule (swift::IRGenDescriptor), (swift::RequestFlags)9>::evaluateRequest(swift::IRGenRequest const&, swift::Evaluator&) + 14
15 swift-frontend           0x000000010e64700f swift::IRGenRequest::OutputType swift::Evaluator::getResultUncached<swift::IRGenRequest, swift::IRGenRequest::OutputType swift::evaluateOrFatal<swift::IRGenRequest>(swift::Evaluator&, swift::IRGenRequest)::'lambda'()>(swift::IRGenRequest const&, swift::IRGenRequest::OutputType swift::evaluateOrFatal<swift::IRGenRequest>(swift::Evaluator&, swift::IRGenRequest)::'lambda'()) + 191
16 swift-frontend           0x000000010e64037c swift::performIRGeneration(swift::FileUnit*, swift::IRGenOptions const&, swift::TBDGenOptions const&, std::__1::unique_ptr<swift::SILModule, std::__1::default_delete<swift::SILModule>>, llvm::StringRef, swift::PrimarySpecificPaths const&, llvm::StringRef, llvm::GlobalVariable**) + 284
17 swift-frontend           0x000000010e0b4dd4 generateIR(swift::IRGenOptions const&, swift::TBDGenOptions const&, std::__1::unique_ptr<swift::SILModule, std::__1::default_delete<swift::SILModule>>, swift::PrimarySpecificPaths const&, llvm::StringRef, llvm::PointerUnion<swift::ModuleDecl*, swift::SourceFile*>, llvm::GlobalVariable*&, llvm::ArrayRef<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>) + 180
18 swift-frontend           0x000000010e0b0293 performCompileStepsPostSILGen(swift::CompilerInstance&, std::__1::unique_ptr<swift::SILModule, std::__1::default_delete<swift::SILModule>>, llvm::PointerUnion<swift::ModuleDecl*, swift::SourceFile*>, swift::PrimarySpecificPaths const&, int&, swift::FrontendObserver*) + 1491
19 swift-frontend           0x000000010e0af4e7 swift::performCompileStepsPostSema(swift::CompilerInstance&, int&, swift::FrontendObserver*) + 727
20 swift-frontend           0x000000010e0c384f withSemanticAnalysis(swift::CompilerInstance&, swift::FrontendObserver*, llvm::function_ref<bool (swift::CompilerInstance&)>, bool) + 143
21 swift-frontend           0x000000010e0b2280 performCompile(swift::CompilerInstance&, int&, swift::FrontendObserver*) + 928
22 swift-frontend           0x000000010e0b0fd1 swift::performFrontend(llvm::ArrayRef<char const*>, char const*, void*, swift::FrontendObserver*) + 3137
23 swift-frontend           0x000000010de8beeb swift::mainEntry(int, char const**) + 2539
24 dyld                     0x00007ff817c1941f start + 1903
Command SwiftCompile failed with a nonzero exit code
```

watchOS:
```
...
4.	While evaluating request IRGenRequest(IR Generation for file "/Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/Sources/NotificationObserver.swift")
5.	While emitting IR SIL function "@$s9CoreStore20NotificationObserverC16notificationName6object5queue7closureACSo014NSNotificationF0a_ypSgSo16NSOperationQueueCSgy10Foundation0C0Vctcfc".
 for 'init(notificationName:object:queue:closure:)' (at /Users/ec2-user/jenkins/workspace/swift-PR-source-compat-suite-debug-macos/swift-source-compat-suite/project_cache/CoreStore/Sources/NotificationObserver.swift:37:5)
Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
0  swift-frontend           0x000000010becfdd8 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) + 40
1  swift-frontend           0x000000010becdf78 llvm::sys::RunSignalHandlers() + 248
2  swift-frontend           0x000000010bed043e SignalHandler(int) + 270
3  libsystem_platform.dylib 0x00007ff817fa05ed _sigtramp + 29
4  libc++abi.dylib          0x00007ff817f2eb7a operator new(unsigned long) + 26
5  swift-frontend           0x0000000105e833a4 swift::irgen::ClassTypeInfo::strongCustomRetain(swift::irgen::IRGenFunction&, swift::irgen::Explosion&, bool) const + 100
6  swift-frontend           0x0000000105e82563 (anonymous namespace)::SyncCallEmission::emitCallToUnmappedExplosion(llvm::CallBase*, swift::irgen::Explosion&) + 531
7  swift-frontend           0x0000000105e6b37d swift::irgen::CallEmission::emitToExplosion(swift::irgen::Explosion&, bool) + 1037
8  swift-frontend           0x000000010607c32b (anonymous namespace)::IRGenSILFunction::visitFullApplySite(swift::FullApplySite) + 4363
9  swift-frontend           0x0000000106056f1c (anonymous namespace)::IRGenSILFunction::visitSILBasicBlock(swift::SILBasicBlock*) + 1692
10 swift-frontend           0x000000010605573d (anonymous namespace)::IRGenSILFunction::emitSILFunction() + 11389
11 swift-frontend           0x000000010605248d swift::irgen::IRGenModule::emitSILFunction(swift::SILFunction*) + 2253
12 swift-frontend           0x0000000105ed1084 swift::irgen::IRGenerator::emitGlobalTopLevel(std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>> const&) + 836
13 swift-frontend           0x0000000105ff6c40 swift::IRGenRequest::evaluate(swift::Evaluator&, swift::IRGenDescriptor) const + 2720
14 swift-frontend           0x0000000106051701 swift::GeneratedModule swift::SimpleRequest<swift::IRGenRequest, swift::GeneratedModule (swift::IRGenDescriptor), (swift::RequestFlags)9>::callDerived<0ul>(swift::Evaluator&, std::__1::integer_sequence<unsigned long, 0ul>) const + 209
15 swift-frontend           0x00000001060515ee swift::SimpleRequest<swift::IRGenRequest, swift::GeneratedModule (swift::IRGenDescriptor), (swift::RequestFlags)9>::evaluateRequest(swift::IRGenRequest const&, swift::Evaluator&) + 14
16 swift-frontend           0x000000010600000f swift::IRGenRequest::OutputType swift::Evaluator::getResultUncached<swift::IRGenRequest, swift::IRGenRequest::OutputType swift::evaluateOrFatal<swift::IRGenRequest>(swift::Evaluator&, swift::IRGenRequest)::'lambda'()>(swift::IRGenRequest const&, swift::IRGenRequest::OutputType swift::evaluateOrFatal<swift::IRGenRequest>(swift::Evaluator&, swift::IRGenRequest)::'lambda'()) + 191
17 swift-frontend           0x0000000105ff937c swift::performIRGeneration(swift::FileUnit*, swift::IRGenOptions const&, swift::TBDGenOptions const&, std::__1::unique_ptr<swift::SILModule, std::__1::default_delete<swift::SILModule>>, llvm::StringRef, swift::PrimarySpecificPaths const&, llvm::StringRef, llvm::GlobalVariable**) + 284
18 swift-frontend           0x0000000105a6ddd4 generateIR(swift::IRGenOptions const&, swift::TBDGenOptions const&, std::__1::unique_ptr<swift::SILModule, std::__1::default_delete<swift::SILModule>>, swift::PrimarySpecificPaths const&, llvm::StringRef, llvm::PointerUnion<swift::ModuleDecl*, swift::SourceFile*>, llvm::GlobalVariable*&, llvm::ArrayRef<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>) + 180
19 swift-frontend           0x0000000105a69293 performCompileStepsPostSILGen(swift::CompilerInstance&, std::__1::unique_ptr<swift::SILModule, std::__1::default_delete<swift::SILModule>>, llvm::PointerUnion<swift::ModuleDecl*, swift::SourceFile*>, swift::PrimarySpecificPaths const&, int&, swift::FrontendObserver*) + 1491
20 swift-frontend           0x0000000105a684e7 swift::performCompileStepsPostSema(swift::CompilerInstance&, int&, swift::FrontendObserver*) + 727
21 swift-frontend           0x0000000105a7c84f withSemanticAnalysis(swift::CompilerInstance&, swift::FrontendObserver*, llvm::function_ref<bool (swift::CompilerInstance&)>, bool) + 143
22 swift-frontend           0x0000000105a6b280 performCompile(swift::CompilerInstance&, int&, swift::FrontendObserver*) + 928
23 swift-frontend           0x0000000105a69fd1 swift::performFrontend(llvm::ArrayRef<char const*>, char const*, void*, swift::FrontendObserver*) + 3137
24 swift-frontend           0x0000000105844eeb swift::mainEntry(int, char const**) + 2539
25 dyld                     0x00007ff817c1941f start + 1903
```

lorentey commented:
@swift-ci test source compatibility

lorentey commented:
@swift-ci test

stephentyrone commented on stdlib/public/core/MutableCollection.swift:
```diff
@@ -529,24 +529,24 @@ extension MutableCollection {
-  Builtin.initialize(tmp, p2)
+  let temp = consume a
+  a = consume b
+  b = consume temp
```
🎉

Azoy commented on test/Prototypes/Hypoarray.swift:
```diff
@@ -0,0 +1,293 @@
+// REQUIRES: executable_test
+
+import StdlibUnittest
+import Synchronization
```
Unused import? This is important because this import requires a `REQUIRES: synchronization` to use.

tshortli commented on stdlib/public/core/Optional.swift:
```diff
@@ -232,7 +232,7 @@ extension Optional where Wrapped: ~Copyable {
     #if compiler(>=6.0) && $NoncopyableGenerics
     switch self {
-    case .some(_borrowing y):
+    case .some(let y):
```
I'm assuming these changes are in reaction to warnings about `_borrowing` being deprecated. I had to change these to use the deprecated spellings intentionally in order to ensure that the standard library's `.swiftinterface` is still buildable with the older compiler versions that we support. Until we drop support for those older compiler versions these warnings cannot be addressed I'm afraid.

lorentey replied:
Local testing on release/6.0 indicates we're good -- those old compilers appear to accept this PR (or rather, the sibling PR #73810) as is, with no changes. Switching back from `_borrowing` to `let` may actually be working in our favor here. Lucky escape!

glessard commented on stdlib/public/core/MutableCollection.swift:
```diff
@@ -529,24 +529,24 @@ extension MutableCollection {
-#endif
-  _debugPrecondition(
-    p1 != p2,
-    "swapping a location with itself is not supported")
```
I love the new formulation (it's finally here!), but we are losing this explanatory assertion. We could have
```
_debugPrecondition(
  Builtin.unprotectedAddressOf(&a) != Builtin.unprotectedAddressOf(&b),
```

lorentey replied:
That's guaranteed in a much better way by Swift's exclusivity rules. I don't think we'd need/want to double check the compiler's work here.

glessard replied:
Good point! I guess this must have only been a hangover from Swift 3, then.

lorentey replied:
Yep! This code originated in 2015, in the Swift 2 era. 🦕

ahoppen commented on test/IDE/complete_unresolved_members.swift:
```diff
@@ -148,7 +148,7 @@ class C4 {
 
 // Exhaustive to make sure we don't include `init({#(some):` or `init({#nilLiteral:` entries
-// UNRESOLVED_3_OPT: Begin completions, 9 items
+// UNRESOLVED_3_OPT: Begin completions, 10 items
```
Could you add the new items to the `-DAG` lines below? Same for the test in complete_generic_optional.swift.

lorentey replied:
D'oh, good catch, thanks! I thought I got them all.

lorentey replied:
I added the missing item to this case, but I left complete_generic_optional.swift as is -- it is not intended to be an exhaustive test.

lorentey committed:
[test] Fix issues uncovered in code review

lorentey committed:
[stdlib] Add missing overload of `extracting` for UnboundedRange

I keep forgetting that `some RangeExpression<Element>` does not cover the `UnboundedRange` form.

lorentey commented:
Temporarily holding until the LSG has a chance to discuss https://github.com/swiftlang/swift-evolution/pull/2489.

lorentey commented:
@swift-ci test

lorentey commented:
@swift-ci build toolchain

lorentey commented:
@swift-ci test source compatibility

lorentey commented:
All known issues have been addressed. This is technically ready to merge pending favorable CI outcome.

However, the PR includes a couple of tiny amendments to SE-0437; I'm temporarily holding off from landing this until the LSG has had a chance to discuss https://github.com/swiftlang/swift-evolution/pull/2489.

lorentey commented on stdlib/public/core/UnsafeBufferPointer.swift.gyb:
```diff
@@ -456,6 +456,98 @@ extension Unsafe${Mutable}BufferPointer where Element: ~Copyable {
+  //
+  /// - Returns: The same buffer as `self`.
+  @_alwaysEmitIntoClient
+  func extracting(_ bounds: UnboundedRange) -> Self {
```
Oh my gosh.

```suggestion
  public func extracting(_ bounds: UnboundedRange) -> Self {
```

lorentey replied:
(I force-pushed to pretend this never happened.)

lorentey commented:
@swift-ci test

lorentey commented:
@swift-ci build toolchain

lorentey commented:
@swift-ci test source compatibility

lorentey commented:
Some of the toolchains builds will fail as #74486 has not landed yet.
